### PR TITLE
Feature/phpunit 7 ugrade

### PR DESCRIPTION
**Pull request by mike-ionut-mihai-sandu-tao**
Updated phpunit to version 7.0.3
tested on : PHP 7.2.24 (cli) (built: Oct 25 2019 04:29:43)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -72,8 +72,8 @@
   },
   "require-dev": {
     "mikey179/vfsstream": "1.4.0",
-    "phpunit/phpunit": "^4.8",
-    "php-mock/php-mock": "^2.0",
+    "phpunit/phpunit": "7.0.*",
```
Comment (julien-sebire) [inline composer.json]: The plan is to put all this in a separate library, right?

File: test/PHPUnit_Framework_ExpectationFailedException.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+namespace oat\generis\test;
+
+use PHPUnit\Framework\ExpectationFailedException;
+
+class PHPUnit_Framework_ExpectationFailedException extends ExpectationFailedException
+{
+
+}
```
Comment (julien-sebire) [inline test/PHPUnit_Framework_ExpectationFailedException.php]: Missing line feed at end of file.

File: test/PHPUnit_Framework_ExpectationFailedException.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\generis\test;
+
+use PHPUnit\Framework\ExpectationFailedException;
+
+class PHPUnit_Framework_ExpectationFailedException extends ExpectationFailedException
```
Comment (julien-sebire) [inline test/PHPUnit_Framework_ExpectationFailedException.php]: Why creating a new class with this kind of non-namespaced name?

File: test/TestCase.php [hunk 4]
```diff
@@ -21,29 +21,30 @@
...
 use common_persistence_sql_dbal_Driver;
 use PHPUnit\Framework\TestCase as UnitTestCase;
-use oat\generis\test\MockObject;
+use \PHPUnit\Framework\MockObject\MockObject;
 
 abstract class TestCase extends UnitTestCase
 {
     /**
-     * Forward compatibility function for PHPUnit 7.0
+     * backward compatibility function for PHPUnit 4.0
```
Comment (julien-sebire) [inline test/TestCase.php]: We're supposed to be in phpunit version 5.7, right?

**Review by julien-sebire — CHANGES_REQUESTED** [review]
I'm not sure this is coordinated with what @shpran already did on this spike. Please synchronize with him...

File: test/TestCase.php [hunk 5]
```diff
@@ -21,29 +21,30 @@
...
      * @param string $exception
      */
-    public function expectException($exception)
+    public function setExpectedException($exception)
```
Comment (Yury-Razhkou) [inline test/TestCase.php]: 4.8.36 :
```
    public function setExpectedException($exceptionName, $exceptionMessage = '', $exceptionCode = null)
{
        $this->expectedException        = $exceptionName;
        $this->expectedExceptionMessage = $exceptionMessage;
        $this->expectedExceptionCode    = $exceptionCode;
    }
```
To keep messages we can adopt this to:
```
public function setExpectedException(string $exception,strin $message='',string $code = '')
    {
        $this->expectException($exception);
        if($message){
        	$this->expectExceptionMessage($message);
        }
        if($code){
        	$this->expectExceptionCode($code);
        }
    }
```

**Pull request closed without merging**